03/3_is_array_overlap/main.c

```diff
@@ -0,0 +1,27 @@
+#include <stdio.h>
+
+#include "memory.h"
+
+int main(void)
+{
+    int nums1[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+    int nums2[5] = { 1, 2, 8, 9, 10 };
+    int* nums3 = nums1 + 2;
+    const size_t NUMS3_LENGTH = 5u;
+
+    printf("Are nums1 and nums2 overlapped?: %s\n",
+        is_overlap(nums1, ARRAY_LENGTH(nums1), nums2, ARRAY_LENGTH(nums2)) == TRUE

```

Comment:
this code is getting too long. so i'd say move 

is_overlap part to a separate line.. assign the return value to char* .. and print that..

also remove (==TRUE) adding this on ternary op sounds weird